clients/admin-ui/src/features/datastore-connections/system_portal_config/ConsentAutomationForm.tsx

```diff
@@ -0,0 +1,267 @@
+    });
+
+    if (isErrorResult(result)) {
+      toast(errorToastParams("Failed to save consent automation"));
+    } else {
+      toast(successToastParams(`Consent automation has been saved`));
+    }
+  };
+
+  useEffect(() => {
+    if (data) {
+      setConsentableItems(data);
+    }
+  }, [data]);
+
+  useEffect(() => {
+    if (notices) {
+      setNoticesOptions(
+        notices.items.map((notice) => ({
+          label: notice.name,
+          value: notice.id,
+        })),
+      );
+    }
+  }, [notices]);
+
+  if (isLoadingConsentableItems || isLoadingNotices) {
+    return (
+      <Box borderWidth="1px" borderRadius="md" {...props}>
+        <Skeleton height="35px" />
+      </Box>
+    );
+  }
+
+  if (!consentableItems || !notices) {
+    return null;
+  }
+
+  const handleNoticeChange = (
+    value: ConsentableItem,
+    parent?: ConsentableItem,
+  ): void => {
+    const updatedItems = consentableItems.map((i) => {
+      if (
+        parent
+          ? i.external_id === parent.external_id
+          : i.external_id === value.external_id
+      ) {
+        if (parent) {
+          return {
+            ...i,
+            children: i.children?.map((child) => {
+              if (child.external_id === value.external_id) {
+                return value;
+              }
+              return child;
+            }),
+          };
+        }
+        return value;
+      }
+      return i;
+    });
+    setConsentableItems(updatedItems);
+  };
+
+  return (
+    <Box borderWidth="1px" borderRadius="md" {...props}>
+      <Accordion allowMultiple data-testid="accordion-consent-automation">
+        <AccordionItem border="none">
+          <AccordionButton>
+            <Box as="span" flex="1" textAlign="left">
+              <Text as="h2" fontWeight="semibold" fontSize="sm">
+                Consent automation
+              </Text>
+            </Box>
+            <AccordionIcon />
+          </AccordionButton>
+          <AccordionPanel
+            p={5}
+            fontSize="sm"
+            data-testid="accordion-panel-consent-automation"
+          >
+            <Text mb={7}>
+              Configure the notices that will trigger consent propagation for
+              this integration. A default value can be used to trigger every
+              consent preference or each item can be configured individually.
+            </Text>
+            <Formik initialValues={initialValues || {}} onSubmit={handleSubmit}>
+              <Form>
+                <SimpleGrid columns={2} spacing={3}>
+                  {consentableItems.map((item) => (
+                    <Fragment key={item.external_id}>
+                      <ConsentableItemField
+                        item={item}
+                        options={noticesOptions}
+                        onNoticeChange={handleNoticeChange}
+                      />
+                      {item.children?.map((child) => (
```

Comment:
does this impl handle `ConsentableItem`s of any depth? E.g if our data looked like:

```
[{
    external_id: "34419",
    type: "Channel",
    name: "Default SMS channel (SMS)",
    notice_id: null,
    children: [
      {
        external_id: "40007",
        type: "Message type",
        name: "Default SMS message type",
        notice_id: null,
        children: [
            {
              external_id: "40001",
              type: "Message type",
              name: "Default SMS message type",
              notice_id: null,
              children: [{etc...}],
              unmapped: true,
      },
        ],
        unmapped: true,
      },
    ],
    unmapped: true,
  },
etc...
]
```

Reply:
It assumes only one level of depth, so that would technically "work" but visually it would only be 1 level deep. Do we need to support multiple levels like this?

Reply:
Let's confirm with @galvana that we can have any `n` number of levels. I'd been building the BE in anticipation of this. 

If so, then we'll need UI guidance on how we should visualize this if > 2 levels total

Reply:
Let's go with one level of depth for UI purposes. @gilluminate is it possible to post back a payload with deeper levels even if they're not exposed in the UI for mapping?

Reply:
The payload gets updated via spread operator, so any deeper children will still remain in tact when posted back.

Reply:
Perfect! Then what we have now is the expected behavior 👍

Reply:
thanks for confirming, all!